src/data/image_upload.ts

```diff
@@ -12,8 +12,19 @@ export interface ImageMutableParams {
   name: string;
 }
 
-export const generateImageThumbnailUrl = (mediaId: string, size: number) =>
-  `/api/image/serve/${mediaId}/${size}x${size}`;
+export const generateImageThumbnailUrl = (
+  mediaId: string,
+  size?: number,
+  original: boolean = false
+) => {
+  if (!original && !size) {
+    throw new Error("Size must be provided if original is false");
+  }
+
+  return original
+    ? `/api/image/serve/${mediaId}/original`
+    : `/api/image/serve/${mediaId}/${size}x${size}`;
```

Comment:
or maybe don't use an `original` parameter, so if `size` is not set, provide the original

```suggestion
  size?: number
) => {
  return size && size > 0
    ? `/api/image/serve/${mediaId}/${size}x${size}`
    : `/api/image/serve/${mediaId}/original`;
```

Reply:
I was thinking about that approach, but in `ha-picture-upload` we use the same structure (original defaulting to false, size defaulting to 512) and I thought it'd be cleaner to keep it unified. Alternatively the original property could be removed in `ha-picture-upload` and the default of size could be changed to 0, but I'm not a fan of changing the default size to an "invalid" one.